src/core/server/core_telemetry/core_telemetry_client.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CORE_TELEMETRY_TYPE } from './constants';
+import { CoreTelemetry } from './types';
+import {
+  ISavedObjectsRepository,
+  SavedObjectsImportOptions,
+  SavedObjectsResolveImportErrorsOptions,
+  SavedObjectsExportOptions,
+} from '..';
+
+/** @internal */
+export type IncrementSavedObjectsImportOptions = Pick<
+  SavedObjectsImportOptions,
+  'createNewCopies' | 'overwrite'
+>;
+/** @internal */
+export type IncrementSavedObjectsResolveImportErrorsOptions = Pick<
+  SavedObjectsResolveImportErrorsOptions,
+  'createNewCopies'
+>;
+/** @internal */
+export type IncrementSavedObjectsExportOptions = Pick<SavedObjectsExportOptions, 'types'> & {
+  supportedTypes: string[];
+};
+
+const SAVED_OBJECTS_IMPORT_DEFAULT = Object.freeze({
+  total: 0,
+  createNewCopies: Object.freeze({ enabled: 0, disabled: 0 }),
+  overwrite: Object.freeze({ enabled: 0, disabled: 0 }),
+});
+const SAVED_OBJECTS_RESOLVE_IMPORT_ERRORS_DEFAULT = Object.freeze({
+  total: 0,
+  createNewCopies: Object.freeze({ enabled: 0, disabled: 0 }),
+});
+const SAVED_OBJECTS_EXPORT_DEFAULT = Object.freeze({
+  total: 0,
+  allTypes: Object.freeze({ yes: 0, no: 0 }),
+});
+
+/** @internal */
+export class CoreTelemetryClient {
+  constructor(
+    private readonly debugLogger: (message: string) => void,
+    private readonly repository: ISavedObjectsRepository
+  ) {}
+
+  public async getTelemetryData() {
+    this.debugLogger('getTelemetryData() called');
+    let coreTelemetry: CoreTelemetry = {};
+    try {
+      const result = await this.repository.get<CoreTelemetry>(
+        CORE_TELEMETRY_TYPE,
+        CORE_TELEMETRY_TYPE
+      );
+      coreTelemetry = result.attributes;
+    } catch (err) {
+      // do nothing
+    }
+    return coreTelemetry;
+  }
+
+  public async incrementSavedObjectsImport({
+    createNewCopies,
+    overwrite,
+  }: IncrementSavedObjectsImportOptions) {
+    const coreTelemetry = await this.getTelemetryData();
```

Comment:
Each of these telemetry client functions has three steps:

1. Retrieve current telemetry data (or an empty attributes object, if no current telemetry data exists)
2. Update the attributes, incrementing whatever counter field(s) are necessary
3. Persist telemetry data to Elasticsearch

There is a race condition if two users call the same API at the same time (either in a single Kibana server, or separate Kibana servers in a HA configuration). However, the impact is minimal, in that edge case we'll simply lose one of the increments. I think that's an acceptable risk.

The alternative is using the document version to ensure the saved object hasn't changed before we update it, and in that scenario, calculate new attributes so we can retry. Seems like overkill to me, though.